`azurerm_storage_object_replication_resource` fails when destination storage account is in a different subscription

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

```
Terraform v1.0.11
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v2.87.0
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_storage_object_replication_resource `

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
provider "azurerm" {
  features {}
}

provider "azurerm" {
  alias = "dev"
  subscription_id = "00000000-0000-0000-0000-000000000000" 
  features {}
}

resource "azurerm_resource_group" "src" {
  name     = "jb-test-src"
  location = "Australia East"
}

resource "azurerm_storage_account" "src" {
  name                     = "srcstoragedc372364"
  resource_group_name      = azurerm_resource_group.src.name
  location                 = azurerm_resource_group.src.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  blob_properties {
    versioning_enabled  = true
    change_feed_enabled = true
  }
}

resource "azurerm_storage_container" "src" {
  name                  = "strcontainer"
  storage_account_name  = azurerm_storage_account.src.name
  container_access_type = "private"
}

resource "azurerm_resource_group" "dst" {
  provider = azurerm.dev

  name     = "jb-test-dst"
  location = "Australia Southeast"
}

resource "azurerm_storage_account" "dst" {
  provider = azurerm.dev

  name                     = "dststoragedc372364"
  resource_group_name      = azurerm_resource_group.dst.name
  location                 = azurerm_resource_group.dst.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
  blob_properties {
    versioning_enabled  = true
    change_feed_enabled = true
  }
}

resource "azurerm_storage_container" "dst" {
  provider = azurerm.dev

  name                  = "strcontainer"
  storage_account_name  = azurerm_storage_account.dst.name
  container_access_type = "private"
}

resource "azurerm_storage_object_replication" "example" {
  source_storage_account_id      = azurerm_storage_account.src.id
  destination_storage_account_id = azurerm_storage_account.dst.id
  rules {
    source_container_name      = azurerm_storage_container.src.name
    destination_container_name = azurerm_storage_container.dst.name
  }
}
```

### Debug Output

```
2021-12-03T10:10:01.755+1300 [INFO]  Starting apply for azurerm_storage_object_replication.example
2021-12-03T10:10:01.758+1300 [DEBUG] azurerm_storage_object_replication.example: applying the planned Create change
2021-12-03T10:10:01.760+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: AzureRM Request: 
GET /subscriptions/[SourceSubscription]/resourceGroups/jb-test-dst/providers/Microsoft.Storage/storageAccounts/dststoragedc372364/objectReplicationPolicies?api-version=2021-04-01 HTTP/1.1
Host: management.azure.com
User-Agent: Go/go1.17.3 (arm64-darwin) go-autorest/v14.2.1 Azure-SDK-For-Go/v59.2.0 storage/2021-04-01 HashiCorp Terraform/1.0.11 (+https://www.terraform.io) Terraform Plugin SDK/2.8.0 terraform-provider-azurerm/2.87.0 pid-222c6c49-1b0a-5959-a213-6608f9eb8820
X-Ms-Correlation-Request-Id: f951d773-1fd9-efc8-ef05-6ac7eb322d5f
Accept-Encoding: gzip: timestamp=2021-12-03T10:10:01.760+1300
2021-12-03T10:10:02.008+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: AzureRM Response for https://management.azure.com/subscriptions/[SourceSubscription]/resourceGroups/jb-test-dst/providers/Microsoft.Storage/storageAccounts/dststoragedc372364/objectReplicationPolicies?api-version=2021-04-01: 
HTTP/2.0 404 Not Found
Content-Length: 103
Cache-Control: no-cache
Content-Type: application/json; charset=utf-8
Date: Thu, 02 Dec 2021 21:10:01 GMT
Expires: -1
Pragma: no-cache
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: f951d773-1fd9-efc8-ef05-6ac7eb322d5f
X-Ms-Failure-Cause: gateway
X-Ms-Request-Id: 229fcffa-c076-4bd8-a100-8fa7d46525eb
X-Ms-Routing-Request-Id: AUSTRALIAEAST:20211202T211002Z:229fcffa-c076-4bd8-a100-8fa7d46525eb

{"error":{"code":"ResourceGroupNotFound","message":"Resource group 'jb-test-dst' could not be found."}}: timestamp=2021-12-03T10:10:02.008+1300
2021-12-03T10:10:02.012+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: PANIC: runtime error: invalid memory address or nil pointer dereference
2021-12-03T10:10:02.012+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: [signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x109178f48]
2021-12-03T10:10:02.012+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 
2021-12-03T10:10:02.012+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: goroutine 290 [running]:
2021-12-03T10:10:02.013+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageObjectReplicationCreate(0x14000accf80, {0x109ddfb80, 0x14001a96a80})
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/storage/storage_object_replication_resource.go:130 +0x288
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000c93180, {0x10a7ad308, 0x1400063b440}, 0x14000accf80, {0x109ddfb80, 0x14001a96a80})
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x170
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000c93180, {0x10a7ad308, 0x1400063b440}, 0x14001a80340, 0x14000accd80, {0x109ddfb80, 0x14001a96a80})
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x8d8
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140010d6300, {0x10a7ad308, 0x1400063b440}, 0x14000742000)
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xe38
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400058c6c0, {0x10a7ad3b0, 0x1400087e5a0}, 0x14000a6fb20)
2021-12-03T10:10:02.014+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:332 +0x94
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10a30dd80, 0x1400058c6c0}, {0x10a7ad3b0, 0x1400087e5a0}, 0x14001ab0480, 0x0)
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c0
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0, 0x1400126a4e0, 0x10ec6b1a0, 0x0)
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc04
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: google.golang.org/grpc.(*Server).handleStream(0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0, 0x0)
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa34
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000dac8f0, 0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0)
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x94
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-12-03T10:10:02.015+1300 [DEBUG] provider.terraform-provider-azurerm_v2.87.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1f0
2021-12-03T10:10:02.024+1300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.87.0/darwin_arm64/terraform-provider-azurerm_v2.87.0_x5 pid=9146 error="exit status 2"
2021-12-03T10:10:02.024+1300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-12-03T10:10:02.025+1300 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = transport is closing"
2021-12-03T10:10:02.077+1300 [DEBUG] provider: plugin exited
```

### Panic Output

```
Stack trace from the terraform-provider-azurerm_v2.87.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x109178f48]

goroutine 290 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageObjectReplicationCreate(0x14000accf80, {0x109ddfb80, 0x14001a96a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/storage/storage_object_replication_resource.go:130 +0x288
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x14000c93180, {0x10a7ad308, 0x1400063b440}, 0x14000accf80, {0x109ddfb80, 0x14001a96a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:318 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000c93180, {0x10a7ad308, 0x1400063b440}, 0x14001a80340, 0x14000accd80, {0x109ddfb80, 0x14001a96a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:456 +0x8d8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140010d6300, {0x10a7ad308, 0x1400063b440}, 0x14000742000)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xe38
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400058c6c0, {0x10a7ad3b0, 0x1400087e5a0}, 0x14000a6fb20)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:332 +0x94
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10a30dd80, 0x1400058c6c0}, {0x10a7ad3b0, 0x1400087e5a0}, 0x14001ab0480, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0, 0x1400126a4e0, 0x10ec6b1a0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc04
google.golang.org/grpc.(*Server).handleStream(0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa34
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x14000dac8f0, 0x14000b36700, {0x10a810958, 0x140012de000}, 0x14001bc97a0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x1f0
```

### Expected Behaviour

The replication policy should be created. It is supported to do replication between storage accounts in different subscription (and tenants).

### Actual Behaviour

Terraform crashed as the destination storage account can't be found. This is because the provider is looking in the source subscription, even though a full ID is provided for the destination account.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

--- a/internal/services/storage/storage_object_replication_resource.go
+++ b/internal/services/storage/storage_object_replication_resource.go
@@ -127,9 +127,11 @@ func resourceStorageObjectReplicationCreate(d *pluginsdk.ResourceData, meta inte
 			return fmt.Errorf("checking for present of existing Storage Object Replication for destination %q): %+v", dstAccount, err)
 		}
 	}
-	for _, existing := range *existingList.Value {
-		if existing.ID != nil && *existing.ID != "" && existing.SourceAccount != nil && *existing.SourceAccount == srcAccount.Name && existing.DestinationAccount != nil && *existing.DestinationAccount == dstAccount.Name {
-			return tf.ImportAsExistsError("azurerm_storage_object_replication", *existing.ID)
+	if existingList.Value != nil {
+		for _, existing := range *existingList.Value {
+			if existing.ID != nil && *existing.ID != "" && existing.SourceAccount != nil && *existing.SourceAccount == srcAccount.Name && existing.DestinationAccount != nil && *existing.DestinationAccount == dstAccount.Name {
+				return tf.ImportAsExistsError("azurerm_storage_object_replication", *existing.ID)
+			}
 		}
 	}
 

